extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java

```diff
@@ -0,0 +1,615 @@
+        // Check the connection to make sure we're not already connected to this device
+        if (connections.containsValue(address)) {
+            return false;
+        }
+
+        // FIXME: When getting here, I always found all connections to be already taken and thus the code never
+        // proceeded. Relaxing this condition did not do any obvious harm, but now guaranteed that the services are
+        // queried from the device.
+        if (connections.size() == maxConnections + 1) {
+            logger.debug("BlueGiga: Attempt to connect to {} but no connections available.", address);
+            return false;
+        }
+
+        bgSetMode();
+
+        // Connect...
+        int connIntervalMin = 60;
+        int connIntervalMax = 100;
+        int latency = 0;
+        int timeout = 100;
+
+        BlueGigaConnectDirectCommand connect = new BlueGigaConnectDirectCommand();
+        connect.setAddress(address.toString());
+        connect.setAddrType(addressType);
+        connect.setConnIntervalMin(connIntervalMin);
+        connect.setConnIntervalMax(connIntervalMax);
+        connect.setLatency(latency);
+        connect.setTimeout(timeout);
+        BlueGigaConnectDirectResponse connectResponse = (BlueGigaConnectDirectResponse) bgHandler
+                .sendTransaction(connect);
+        if (connectResponse.getResult() != BgApiResponse.SUCCESS) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Close a connection using {@link BlueGigaDisconnectCommand}
+     *
+     * @param connectionHandle
+     * @return
+     */
+    public boolean bgDisconnect(int connectionHandle) {
+        BlueGigaDisconnectCommand command = new BlueGigaDisconnectCommand();
+        command.setConnection(connectionHandle);
+        BlueGigaDisconnectResponse response = (BlueGigaDisconnectResponse) bgHandler.sendTransaction(command);
+
+        return response.getResult() == BgApiResponse.SUCCESS;
+    }
+
+    /**
+     * Device discovered. This simply passes the discover information to the discovery service for processing.
+     */
+    public void deviceDiscovered(BluetoothDevice device) {
+        if (discoveryActive) {
+            for (BluetoothDiscoveryListener listener : discoveryListeners) {
+                listener.deviceDiscovered(device);
+            }
+        }
+    }
+
+    /**
+     * Start a read of all primary services using {@link BlueGigaReadByGroupTypeCommand}
+     *
+     * @param connectionHandle
+     * @return true if successful
+     */
+    public boolean bgFindPrimaryServices(int connectionHandle) {
+        logger.debug("BlueGiga FindPrimary: connection {}", connectionHandle);
+        BlueGigaReadByGroupTypeCommand command = new BlueGigaReadByGroupTypeCommand();
+        command.setConnection(connectionHandle);
+        command.setStart(1);
+        command.setEnd(65535);
+        command.setUuid(UUID.fromString("00002800-0000-0000-0000-000000000000"));
```

Comment:
This looks like a perfect candidate for a constant so this magic string gets a human readable meaning, doesn't it?

Reply:
Did you see this comment?